.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,87 @@
+name: facebook/metro/build-and-deploy
+on:
+  push:
+    branches:
+    - main
+jobs:
+  run-js-checks:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4.1.0
+    - uses: actions/setup-node@v4
+      with:
+        node-version: 18.12
+        cache: yarn
+    - name: Install Dependencies
+      run: npm install --force flow-bin
+    - run: yarn typecheck
+    - run: yarn typecheck-ts
+    - run: yarn lint
+    - run: yarn test-smoke
+  test-with-coverage:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4.1.0
+    - uses: actions/setup-node@v4
+      with:
+        node-version: 18.12
+        cache: yarn
+    - name: Install Dependencies
+      run: npm install --force @babel/core
```

Comment:
We need to be using Yarn to install dependencies so that `yarn.lock` is respected - that's the reason for the test failure you're seeing.

I'm not sure what the significance of `@babel/core` is here (and similarly `--save-dev jest` in jobs below)? `yarn --frozen-lockfile --non-interactive --ignore-scripts` is the corresponding line in our CircleCI `config.yml`, and we use that for all jobs:

https://github.com/facebook/metro/blob/59a0e366f5554f99b1d04b0760e40a969a54b722/.circleci/config.yml#L35